src/praisonai-agents/praisonaiagents/agent/handoff.py

```
@@ -405,6 +405,93 @@ def _compute_effective_tools(self, source_agent: 'Agent') -> List[Any] | None:
+            logger.debug(f"Resolving runtime for handoff: agent_id={target_agent_id}, model={target_model_ref}")
+            
+            # Resolve runtime for target agent
+            runtime = resolve_runtime(target_agent_id, target_model_ref, session_ctx)
+            
+            # Execute using resolved runtime
+            response = runtime.execute(prompt, tools=effective_tools)
+            logger.info(f"Handoff executed with turn-time resolved runtime: {runtime.provider}/{runtime.model_ref}")
+            
+            return response
+            
+        except Exception as e:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)